src/librustc_typeck/check/method/probe.rs

```diff
@@ -990,6 +1032,39 @@ impl<'a, 'gcx, 'tcx> ProbeContext<'a, 'gcx, 'tcx> {
         })
     }
 
+    fn emit_unstable_name_collision_hint(
+        &self,
+        stable_pick: &Pick,
+        unstable_candidates: &[(&Candidate<'tcx>, Symbol)],
+    ) {
+        let mut diag = self.tcx.struct_span_lint_node(
+            lint::builtin::UNSTABLE_NAME_COLLISION,
+            self.fcx.body_id,
+            self.span,
+            "a method with this name will be added to the standard library in the future",
```

Comment:
it might be good to clarify "and it will take precedence over the method you're calling now" somewhere (maybe not in this string)

Reply:
well I guess the `help` below makes it fairly clear

Reply:
@nikomatsakis Unfortunately no, it won't take precedence. It will cause error E0034 "multiple applicable items in scope" instead.

Reply:
Well, I guess that probably depends, right? (That is, if an inherent method is added, it might take precedence over a trait method...) But ok I see that it's not generally true.